zingolib/src/wallet/transaction_record.rs

```diff
@@ -0,0 +1,343 @@
+use incrementalmerkletree::witness::IncrementalWitness;
+use zcash_primitives::{sapling, transaction::TxId};
+
+use crate::error::ZingoLibError;
+
+use super::{
+    data::{OutgoingTxData, PoolNullifier},
+    *,
+};
+
+///  Everything (SOMETHING) about a transaction
+#[derive(Debug)]
+pub struct TransactionRecord {
+    // the relationship of the transaction to the blockchain. can be either Broadcast (to mempool}, or Confirmed.
+    pub status: ConfirmationStatus,
+
+    // Timestamp of Tx. Added in v4
+    pub datetime: u64,
+
+    // Txid of this transaction. It's duplicated here (It is also the Key in the HashMap that points to this
+    // WalletTx in LightWallet::txs)
+    pub txid: TxId,
+
+    // List of all nullifiers spent in this Tx. These nullifiers belong to the wallet.
```

Comment:
Clarify comment to: "...all nullifiers spent in this transaction by this account..." to account for the fact that we won't have any data for incoming transactions.